simulationdataschema/atoms_state.py

```diff
@@ -0,0 +1,571 @@
+        """
+        Resolves the occupation of the orbital state. The occupation is resolved from the degeneracy
+        and the number of excited electrons.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+        """
+        if self.orbital_ref is None or self.n_excited_electrons is None:
+            logger.warning(
+                "Cannot resolve occupation without `orbital_ref` or `n_excited_electrons`."
+            )
+            return
+        if self.orbital_ref.occupation is None:
+            degeneracy = self.orbital_ref.resolve_degeneracy()
+            if degeneracy is None:
+                logger.warning("Cannot resolve occupation without `degeneracy`.")
+                return
+            self.orbital_ref.occupation = degeneracy - self.n_excited_electrons
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Check if n_excited_electrons is between 0 and 1
+        if 0.0 <= self.n_excited_electrons <= 1.0:
+            logger.error("Number of excited electrons must be between 0 and 1.")
+
+        # If dscf_state is 'initial', then n_excited_electrons is set to 0
+        if self.dscf_state == "initial":
+            self.n_excited_electrons = None
+            self.degeneracy = 1
+
+        # Resolve the occupation of the active orbital state
+        if self.orbital_ref is not None and self.n_excited_electrons:
+            if self.orbital_ref.occupation is None:
+                self.resolve_occupation(logger)
+
+
+class HubbardInteractions(ArchiveSection):
+    """
+    A base section to define the Hubbard interactions of the system.
+    """
+
+    orbitals_ref = Quantity(
+        type=OrbitalsState,
+        shape=["*"],
+        description="""
+        Reference to the `OrbitalsState` sections that are used as a basis to obtain the Hubbard
+        interaction matrices.
+        """,
+    )
+
+    umn = Quantity(
+        type=np.float64,
+        shape=["*", "*"],
+        unit="joule",
+        description="""
+        Value of the local Hubbard interaction matrix. The order of the rows and columns coincide
+        with the elements in `orbital_ref`.
+        """,
+    )
+
+    u = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Value of the (intraorbital) Hubbard interaction
+        """,
+        a_eln=ELNAnnotation(component="NumberEditQuantity"),
+    )
+
+    jh = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Value of the (interorbital) Hund's coupling.
+        """,
+        a_eln=ELNAnnotation(component="NumberEditQuantity"),
+    )
+
+    up = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Value of the (interorbital) Coulomb interaction. In rotational invariant systems, up = u - 2 * jh.
+        """,
+        a_eln=ELNAnnotation(component="NumberEditQuantity"),
+    )
```

Comment:
Same remark as above: `up` is especially confusing. It will give a lot of false hits when performing text searches.
Even `u_h` would already be clearer.